schema.ts

```diff
@@ -1201,7 +1211,24 @@ export interface ObjectSchemaDefinition<K extends string, L extends string>
    * @hidden
    */
   modifiedByProperty?: PropertyIdentifier<K>;
-
+  /**
+   * For cases where the object being synced represents a user, the name of the property within
+   * {@link ObjectSchemaDefinition.properties} that identifies the email address of the user.
+   *
+   * Must be a {@link ValueType.String} property with the {@link ValueHintType.Email} hint or
+   * a {@link ValueType.Object} with the {@link ValueHintType.Person} hint
+   * @hidden
+   */
+  userEmailProperty?: PropertyIdentifier<K>;
+  /**
+   * For cases where the object being synced represents a user, the name of the property within
+   * {@link ObjectSchemaDefinition.properties} that identifies the id of the user in the service
+   * being synced from.
+   *
+   * Must be a {@link ValueType.String} or {@link ValueType.Number} property
+   * @hidden
+   */
+  userIdProperty?: PropertyIdentifier<K>;
```

Comment:
Would it be safe to assume / require that the `idProperty` of the schema of a table with the `Users` role be the user ID? Hard to imagine a case where a Pack maker would want / need a different ID for the row.

Reply:
I was not positive here. There can be multiple uniqueish identifiers in a system. For example in github right now we return the login id (not the numeric id) as the id of the users table https://github.com/coda/packs/blob/main/packs/github/main/schemas.ts. I would think generally for our purposes we would prefer to use the numeric id although it could be argued either way.

Reply:
Yeah that's interesting, it does seem like the idProperty should always be the user id in a users table. I'd be open to omitting this and adding it back if it comes up for real.

Reply:
Would it be reasonable to keep this in but leave it optional, and if not present, assume the `idProperty` is the `userIdProperty`?